migrate to recipe v1 + update to 4.2

claudiushaag opened the pull request:
This PR migrates the feedstock to v1 recipe and carries out the update to v4.2, and rerenders the feedstock.

This would supersede both https://github.com/conda-forge/rarfile-feedstock/pull/6 and https://github.com/conda-forge/rarfile-feedstock/pull/9, as the PR by @ShalokShalom did not include the correct noarch matchspecs for python and declared an external pixi environment, which is not needed. Thanks for providing a basis though!

@marcelotrevisani thank you for reviewing!

<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

claudiushaag committed:
Start with v1 recipe migration

claudiushaag committed:
MNT: Re-rendered with conda-smithy 3.53.3 and conda-forge-pinning 2025.12.15.10.28.39

Other tools:
- conda-build 25.11.0
- rattler-build 0.53.0
- rattler-build-conda-compat 1.4.10

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/recipe.yaml```) and found it was in an excellent condition.

claudiushaag committed:
MNT: Re-rendered with conda-smithy 3.53.3 and conda-forge-pinning 2025.12.15.10.28.39

Other tools:
- conda-build 25.11.0
- rattler-build 0.53.0
- rattler-build-conda-compat 1.4.10
